### DISCUSSION TITLE
How to split a input.i file into seperate parts for better mangement?

### DISCUSSION THREAD

**Discussion by ZhigangPu** [idaholab/moose#19306] [Q&A General] [answered]
Dear,

As my project grows larger, my input file has hundreds of lines.  It becomes more and more complicated to manage.

I wonder is there any way to split a input file into some smaller input file containing different parts? Then we can "include" different parts into a "main" input file to assemble them together. I think that is a good way to manage larger projects.

Thanks
zhigang

**Comment by loganharbour** [comment, marked as answer]
This is currently a work in progress. See #9755 for more information.

You can also use the Action System (see https://mooseframework.inl.gov/source/actions/Action.html; sorry the documentation isn't very useful at the moment) to systematically build objects from more simplified input.

**Reply by GiudGiud** [reply]
includes are still a WIP. 
Actions are a great idea for your application, to simplify the syntax for users.

Another option is to split the input file into multiple files and provide them on the command line like this:
./my-app-opt -i file1.i file2.i ...
the files are then merged together

**Reply by ZhigangPu** [reply]
thanks very much! I'll try that!

**Reply by Traiwit** [reply]
> ./my-app-opt -i file1.i file2.i ...
> the files are then merged together

Hi @GiudGiud, so we can have a separated object for each file and run them together in a single line? for example `./my-app-opt -i mesh.i controls.i constraints.i` ??



a quick note: @loganharbour @GiudGiud This 'include' would be very helpful for our workflow too, as each of our full-scale simulation objects (BCs, Materials, Controls) has more than 2000 lines, not to mention 100000+ lines from nodal constraints.

Cannot wait to see it :)

Thanks guys!

Kind regards,
Traiwit

**Reply by GiudGiud** [reply]
@Traiwit as was discussed on the other post, 100k lines input is problematic. Please make sure to use an action or a different kind of constraint if you don't want to spend dozens of hours just parsing the input file at startup. 

I actually have not tried more than two input files merged. I ll try tomorrow.

**Reply by GiudGiud** [reply]
Yeah so it works to merge more than 2 inputs. Not sure why I couldnt remember that yesterday.

**Reply by Traiwit** [reply]
@GiudGiud Awesome! I tried that on my simulation and it works fine too, thanks!
